main.tf

```diff
@@ -29,6 +29,13 @@ resource "aws_iam_role_policy" "task_execution" {
   policy = "${element(concat(data.aws_iam_policy_document.task_execution_permissions.*.json, list("")), 0)}"
 }
 
+resource "aws_iam_role_policy" "read_ssm_parameter" {
+  count  = "${var.create ? 1 : 0}"
+  name   = "${var.name_prefix}-read-ssm-params-permissions"
+  role   = "${element(concat(aws_iam_role.execution.*.id, list("")), 0)}"
+  policy = "${element(concat(data.aws_iam_policy_document.task_parameter_permissions.*.json, list("")), 0)}"
```

Comment:
@c19 None of the `data` blocks have `count` attributes, so this needs to be

```
policy = "${data.aws_iam_policy_document.task_parameter_permissions.json}"
```

Reply:
In other words, `data.aws_iam_policy_document.task_parameter_permissions.*.json` is invalid, since the `data` block doesn't actually result in a list, since it has no `count`